packages/salesforcedx-sobjects-faux-generator/src/describe/sObjectDescribe.ts

```diff
@@ -244,50 +214,60 @@ export class SObjectDescribe {
       return Promise.reject(e);
     }
     try {
-      const sobjects = JSON.parse(result).result as string[];
+      const sobjects = extractJsonObject(result).result as string[];
       return Promise.resolve(sobjects);
     } catch (e) {
       return Promise.reject(result);
     }
   }
 
-  public async describeSObjectBatch(
-    projectPath: string,
-    types: string[],
-    nextToProcess: number,
-    username?: string
-  ): Promise<SObject[]> {
-    const batchSize = 25;
+  public buildSObjectDescribeURL(sObjectName: string): string {
+    const urlElements = [
+      this.getVersion(),
+      this.sobjectsPart,
+      sObjectName,
+      'describe'
+    ];
+    return urlElements.join('/');
+  }
 
-    await this.setupConnection(projectPath, username);
+  public buildBatchRequestURL(): string {
+    const batchUrlElements = [
+      this.instanceUrl,
+      this.servicesPath,
+      this.getVersion(),
+      this.batchPart
+    ];
+    return batchUrlElements.join('/');
+  }
 
+  public buildBatchRequestBody(
+    types: string[],
+    nextToProcess: number
+  ): BatchRequest {
+    const batchSize = 25;
     const batchRequest: BatchRequest = { batchRequests: [] };
 
     for (
       let i = nextToProcess;
       i < nextToProcess + batchSize && i < types.length;
       i++
     ) {
-      const urlElements = [
-        this.getVersion(),
-        this.sobjectsPart,
-        types[i],
-        'describe'
-      ];
-      const requestUrl = urlElements.join('/');
-
-      batchRequest.batchRequests.push({ method: 'GET', url: requestUrl });
+      batchRequest.batchRequests.push({
+        method: 'GET',
+        url: this.buildSObjectDescribeURL(types[i])
+      });
     }
-    const batchUrlElements = [
-      this.instanceUrl,
-      this.servicesPath,
-      this.getVersion(),
-      this.batchPart
-    ];
-    const batchRequestUrl = batchUrlElements.join('/');
-    const options: XHROptions = {
+
+    return batchRequest;
+  }
+
+  public buildXHROptions(types: string[], nextToProcess: number): XHROptions {
```

Comment:
What does XHR stand for?

Reply:
XMLHttpRequest, we use these objects & library cause we're running describes directly against the org vs running it via the CLI. Here's more info on XHR https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest